sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -1396,39 +1446,136 @@ trait ParquetPartitionReaderBase extends Logging with ScanWithMetrics
    * @return updated block metadata corresponding to the output
    */
   protected def copyBlocksData(
-      in: FSDataInputStream,
+      filePath: Path,
       out: HostMemoryOutputStream,
       blocks: Seq[BlockMetaData],
-      realStartOffset: Long): Seq[BlockMetaData] = {
-    val copyRanges = new ArrayBuffer[CopyRange]
-    val outputBlocks = computeBlockMetaData(blocks, realStartOffset, Some(copyRanges))
-    val copyBuffer = new Array[Byte](copyBufferSize)
-    copyRanges.foreach(copyRange => copyDataRange(copyRange, in, out, copyBuffer))
-    outputBlocks
+      realStartOffset: Long,
+      metrics: Map[String, GpuMetric]): Seq[BlockMetaData] = {
+    val startPos = out.getPos
+    val filePathString: String = filePath.toString
+    val remoteItems = new ArrayBuffer[CopyRange](blocks.length)
+    var totalBytesToCopy = 0L
+    withResource(new ArrayBuffer[LocalCopy](blocks.length)) { localItems =>
+      blocks.foreach { block =>
+        block.getColumns.asScala.foreach { column =>
+          val columnSize = column.getTotalSize
+          val outputOffset = totalBytesToCopy + startPos
+          val channel = FileCache.get.getDataRangeChannel(filePathString,
```

Comment:
Do we need to worry about having too many channels open at once?  Possibly in a follow on issue?

Reply:
Chatted with @revans2 offline and this is not an issue.